plugins/hotkeyshortcuts.koplugin/main.lua

```diff
@@ -0,0 +1,574 @@
+            else
+                readerconfig.key_events.ShowConfigMenu = { { { "Press", "AA" } }, event = "ShowConfigMenu" }
+            end
+            logger.dbg("Hotkey ReaderConfig:registerKeyEvents() overridden.")
+        end
+
+        local readerlink = self.ui.link
+        readerlink.key_events = {} -- reset it.
+        if Device:hasScreenKB() or Device:hasSymKey() then
+            readerlink.key_events.GotoSelectedPageLink = { { "Press" }, event = "GotoSelectedPageLink" }
+        elseif Device:hasKeyboard() then
+            readerlink.key_events = {
+                SelectNextPageLink = {
+                    { "Tab" },
+                    event = "SelectNextPageLink",
+                },
+                SelectPrevPageLink = {
+                    { "Shift", "Tab" },
+                    event = "SelectPrevPageLink",
+                },
+                GotoSelectedPageLink = {
+                    { "Press" },
+                    event = "GotoSelectedPageLink",
+                },
+            }
+        end
+        logger.dbg("Hotkey ReaderLink:registerKeyEvents() overridden.")
+
+        if Device:hasKeyboard() then
+            local readersearch = self.ui.search
+            readersearch.key_events = {} -- reset it.
+            readersearch.key_events.ShowFulltextSearchInputBlank = {
+                { "Alt", "Shift", "S" }, { "Ctrl", "Shift", "S" },
+                event = "ShowFulltextSearchInput",
+                args = ""
+            }
+            logger.dbg("Hotkey ReaderSearch:registerKeyEvents() overridden.")
+        end
+
+        self.ui.toc.key_events = {} -- reset it.
+        logger.dbg("Hotkey ReaderToc:registerKeyEvents() overridden.")
+
+        self.ui.thumbnail.key_events = {} -- reset it.
+        logger.dbg("Hotkey ReaderThumbnail:registerKeyEvents() overridden.")
+
+        local readerui = self.ui
+        readerui.key_events = {} -- reset it, then add our own
+        readerui.key_events.Home = { { "Home" } }
+        readerui.key_events.Back = { { Device.input.group.Back } }
+        if Device:hasDPad() and Device:useDPadAsActionKeys() then
+            readerui.key_events.KeyContentSelection = { { { "Up", "Down" } }, event = "StartHighlightIndicator" }
+        elseif Device:hasKeyboard() then
+            readerui.key_events.Reload = { { "F5" } }
+        end
+        logger.dbg("Hotkey ReaderUI:registerKeyEvents() overridden.")
+    end
+
+    if Device:hasKeyboard() then
+        self.ui.dictionary.key_events = {} -- reset it.
+        logger.dbg("Hotkey ReaderDictionary:registerKeyEvents() overridden.")
+
+        self.ui.wikipedia.key_events = {} -- reset it.
+        logger.dbg("Hotkey ReaderWikipedia:registerKeyEvents() overridden.")
+
+        local filesearcher = self.ui.filesearcher
+        filesearcher.key_events = {} -- reset it.
+
+        filesearcher.key_events.ShowFileSearchBlank = {
+            { "Alt", "Shift", "F" }, { "Ctrl", "Shift", "F" },
```

Comment:
No blank line needed, it's the same object you're tweaking.

What's the reasoning for resetting it, while still keeping the Alt+Shift+F ? The fact that this plugin can't override them ?
I guess it would feel inconsistent if one assign ShowFileSearch to Alt+S, but have to use Alt+Shift+F for ShowFileSearchBlank. Can one assign ShowFileSearchBlank to Alt+R ?
Just asking, no change/fix requested.

Reply:
the reason is that, this is the option with the blank field. Remember we had two? one was prepopulated with the previous search and this one was blank. essentially same as in readersearch.

Reply:
I remembered it.
Re-read my comment please :)

Reply:
the reason is that i still want to have a blank canvas and cannot get it via dispatcher.

> Can one assign ShowFileSearchBlank to Alt+R ?

no, because that does not exist in the dispatcher world... nor it should really